File: frontend-mealsonwheels/src/types/user/PartnerDTO.ts
// src/types/user/PartnerDTO.ts

import type { UserDTO } from "./UserDTO";
import type { DishDTO } from "../meal/DishDTO";
import type { MealDTO } from "../meal/MealDTO";
import type { MenuDTO } from "../meal/MenuDTO";

export interface PartnerDTO extends UserDTO {
  companyName: string;
  companyDescription: string;
  companyAddress: string;
  companyLocationLat: number;
  companyLocationLong: number;
  dishes: DishDTO[];
  providedMeals: MealDTO[];
  menus: MenuDTO[];
}

File: backend-mealsonwheels/src/main/java/com/merrymeal/mealsonwheels/dto copy done/roleDTOs/PartnerDTO.java
package com.merrymeal.mealsonwheels_backend.dto.roleDTOs;

import lombok.*;
import lombok.experimental.SuperBuilder;

import java.util.List;
import com.merrymeal.mealsonwheels_backend.dto.UserDTO;
import com.merrymeal.mealsonwheels_backend.dto.mealDTOs.DishDTO;
import com.merrymeal.mealsonwheels_backend.dto.mealDTOs.MealDTO;
import com.merrymeal.mealsonwheels_backend.dto.mealDTOs.MenuDTO;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@SuperBuilder
public class PartnerDTO extends UserDTO {
    private String companyName;
    private String companyDescription;
    private String companyAddress;
    private double companyLocationLat;
    private double companyLocationLong;
    private List<DishDTO> dishes;
    private List<MealDTO> providedMeals;
    private List<MenuDTO> menus;
}